Update packages, arm template, Cosmos SDK code

markjbrown opened the pull request:
## Purpose
Project couple years out of date. Uses deprecated packages, etc. Needed a refresh.

## Does this introduce a breaking change?
<!-- Mark one with an "x". -->
```
[X] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[X] Bugfix
[ ] Feature
[X] Code style update (formatting, local variables)
[X] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[X] Other... Please describe: Most of the code updated here is Cosmos SDK specific. Although I did version the SimpleEvent class to make it work better with Cosmos and strong typing. Pretty much left everything else alone.
```

markjbrown committed:
Updated template. Update to .NET Core 3.1

markjbrown committed:
updated everything

markjbrown committed:
update readme

scm-xbox commented on TelemetryAPI/TelemetryAPI.csproj:
```diff
@@ -1,38 +1,23 @@
-    <PackageReference Include="Microsoft.Azure.WebJobs.Logging" Version="3.0.5" />
-    <PackageReference Include="Microsoft.IO.RecyclableMemoryStream" Version="1.2.2" />
-    <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="1.0.26" />
+    <Content Remove="C:\Users\mjbrown\.nuget\packages\microsoft.azure.cosmos\3.16.0\contentFiles\any\netstandard2.0\ThirdPartyNotice.txt" />
```
This line shouldn't include your current machine.

markjbrown replied:
fixed.

scm-xbox commented on TelemetryAPI/SimpleEvent.cs:
```diff
@@ -8,6 +8,82 @@
 namespace TelemetryAPI
 {
     public class SimpleEvent
+    {
```
I don't think it's necessarily correct to change the original event definition and rename the old version to V1.  If you do name it V1, then this one should be V2.

Also, it looks like you only added a couple of fields, so you could probably just extend the original event with the new fields.  Schemas like this are fine with extending, its removing fields that can be an issue.

markjbrown replied:
Resolved in latest commit.

markjbrown replied:
fixed.

scm-xbox commented on TelemetryAPI/CosmosReader.cs:
```diff
@@ -16,10 +18,8 @@ namespace TelemetryAPI
-        static readonly FeedOptions feedOptions = new FeedOptions() { MaxItemCount = 30000, EnableCrossPartitionQuery = true };
-        static readonly string _collection = Environment.GetEnvironmentVariable("CosmosDataCollection") ?? "SimpleEvents";
-        static readonly string _database = Environment.GetEnvironmentVariable("CosmosDatabaseId") ?? "TelemetryDB";
+        static CosmosClient cosmosClient = new CosmosClient(Config.CosmosDbUri, Config.CosmosDbAuthKey);
```
The naming convention for these fields lost the _ in front... its used elsewhere in the project.  

Can you either fix the other instances of _ in other files, or use _ in this case for consistency?

markjbrown replied:
fixed and consistent for both reader and sender

scm-xbox reviewed (changes requested):
Overall looks good, but I had a few comments I'd like to see addressed before the final checkin.  Thanks!

markjbrown committed:
Updates to resolve suggestions.

markjbrown commented:
Updated for all suggestions. Also added some perf improvements and a few lines calling out request options that can help tune high volume cross partition queries

thanks.

markjbrown commented:
Sorry, forgot to update to .NET 5. Will do that now and send another commit.

markjbrown commented:
Tried .NET 5 upgrade but got runtime errors from functions.

This is ready for review/merge.

Thanks.
